orchestrator/crates/orchestrator/src/tests/jobs/da_job/mod.rs

```
@@ -126,6 +132,17 @@ async fn test_da_job_process_job_failure_on_pending_block() {
         then.status(200).body(serde_json::to_vec(&response).unwrap());
     });
 
+    // Create proper metadata structure
+    let block_number = internal_id.parse::<u64>().unwrap_or(0);
```

It's not clear why this would fail (and trigger `unwrap_or(0)`). If it's an expected condition, maybe it should have some documentation/helper fn, and if it's not then I would fail hard instead of silently converting to 0.

---

Maybe this is related to handling the genisis block? If so, I would leave a log message so it would be more obvious if this happens accidentally.

---

You are right, there's no benefit of using `unwrap_or` here, have defaulted it to `unwrap`.
And no this is not related to handling the genesis block.

---

Removed it now